source/actions3/actions/AMScanAction.h

```diff
@@ -82,46 +135,129 @@ protected slots:
 	// Slots that handle the state transitions of the scan controller.
 	////////////////////////////////////////////////////////////////////////
 
-	/// Starts the scan controller if it was able to successfully initialize.
+	/*!
+	  * Handles the controller signalling that it has initialized. Attempts to
+	  * transition the controller to the running state.
+	  */
 	void onControllerInitialized();
-	/// Saves the scan to the database if the controller was able to successfully start.
+
+	/*!
+	  * Handles the controller signalling that it has successfully started. Saves
+	  * the scan to the database.
+	  */
 	void onControllerStarted();
-	/// Handles at the action level if the controller is cancelled.
+
+	/*!
+	  * Handles the controller signalling that the scan has been cancelled. If the
+	  * scan is known to the database, then an attempt to resave it is made. Adapts
+	  * the cancelled instruction to the Action3 interface.
+	  */
 	void onControllerCancelled();
-	/// Handles at the action level if the controller fails.
+
+	/*!
+	  * Handles the controller signalling that the scan has failed. If the scan
+	  * is known to the database, then an attempt to resave it is made. Adapts the
+	  * failed instruction to the Action3 interface.
+	  */
 	void onControllerFailed();
-	/// Handles all the wrap up and clean up if the controller succeeds.
+
+	/*!
+	  * Handles the controller signalling that the scan has succeeded. Saves the
+	  * finshed scan to the database and, if the configuration specifies an auto-export,
+	  * then an export is performed. Adapts the succeeded instruction to the Aciton3
+	  * interface.
+	  */
 	void onControllerSucceeded();
-	/// Handles setting the status text when the initialization actions are started.
+
+	/*!
+	  * Handles the controller signalling that it has begun the initialization
+	  * steps. Sets the status text for the action to "Initializing".
+	  */
 	void onControllerInitializing();
-	/// Handles setting the status text when cleaning actions are started.
+
+	/*!
+	  * Handles the controller signalling that it has begun the clean-up steps.
+	  * Sets the status text for the aciton to "Cleaning up".
+	  */
 	void onControllerCleaningUp();
-	/// Handles the progress changed passed up from the controller.
+
+	/*!
+	  * Handles the controller signaling a periodic status changed update. Re-emits
+	  * the progress changed signal on the Action3 interface.
+	  */
 	void onControllerProgressChanged(double elapsed, double total);
-	/// Helper slot that updates the status text when the controller changes state.
+
+	/*!
+	  * Handles the controller signaling that it has made a state transition.
+	  * Updates the status text to reflect the new state.
+	  *
+	  * /// Is this still the case. There's nothing in the body of this method ///
+	  */
```

Comment:
I just copied the comment for this one, as the body for the method is empty. See [here](https://github.com/acquaman/acquaman/blob/master/source/actions3/actions/AMScanAction.cpp#L432-L435).